tasks/generate/cache.rake

```diff
@@ -0,0 +1,57 @@
+def lexer_cache_source(filename, lexer)
+  const_name = lexer.name.split('::').last.to_sym
+  root = Pathname.new(Rouge::ROOT)
+  demo_file = lexer.demo_file.relative_path_from(root).to_s
+
+  # no need to pre-cache the default
+  if demo_file == "lib/rouge/demos/#{lexer.tag}"
+    demo_file = nil
+  end
+
+  yield "  Lexer.cache #{filename.inspect}, #{const_name.inspect}, #{lexer.tag.inspect} do"
+  yield "    @tag = #{lexer.tag.inspect}"
+  yield "    @title = #{lexer.title.inspect}"
+  yield "    @desc = #{lexer.desc.inspect}"
+  yield "    @option_docs = #{lexer.option_docs.to_hash.inspect}" if lexer.option_docs.any?
+  yield "    @demo_file = Pathname.new(#{demo_file.inspect})" if demo_file
+  yield "    @aliases = #{lexer.aliases.inspect}" if lexer.aliases.any?
+  yield "    @filenames = #{lexer.filenames.inspect}" if lexer.filenames.any?
+  yield "    @mimetypes = #{lexer.mimetypes.inspect}" if lexer.mimetypes.any?
+
+  if lexer.detectable?
+    yield lexer.method(:detect?).source
+    yield "    @detectable = true"
+  else
+    yield "    @detectable = false"
+  end
+
+  yield "  end"
+end
+
+namespace :generate do
+  cache_file = './lib/rouge/langspec_cache.rb'
+
+  desc "Update the language cache file"
+  task :cache do
+    sh "echo '# noop' > #{cache_file}"
+    require 'rouge'
+    require 'method_source'
+
+    File.open(cache_file, 'w') do |out|
+      out.puts "# -*- coding: utf-8 -*- #"
+      out.puts "# frozen_string_literal: true"
+      out.puts "# automatically generated by #{__FILE__}"
```

Comment:
Using `__FILE__` here leads to returning an absolute path (refer the cache file committed to this branch.)
Instead, hardcode to avoid unnecessary diffs (when a different maintainer updates the cache):

```suggestion
      out.puts "# automatically generated by 'tasks/generate/cache.rake'"
```

---

<br/>

or to a filepath agnostic comment:
```suggestion
      out.puts "# automatically generated by running `rake generate:cache`"
```